core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -291,24 +374,31 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_3(
 
 static int secp256k1_ecdh_hash_passthrough(uint8_t *output, const uint8_t *x,
                                            const uint8_t *y, void *data) {
-  output[0] = 0x04;
+  output[0] = SECP256K1_TAG_PUBKEY_UNCOMPRESSED;
   memcpy(&output[1], x, 32);
   memcpy(&output[33], y, 32);
   (void)data;
   return 1;
 }
 
-/// def multiply(self, secret_key: bytes, public_key: bytes) -> bytes:
+/// def multiply(
+///     self, secret_key: bytes, public_key: bytes, compressed_result: bool =
+///     False
+/// ) -> bytes:
 ///     """
 ///     Multiplies point defined by public_key with scalar defined by
-///     secret_key. Useful for ECDH.
+///     secret_key. Useful for ECDH. The resulting point is serialized in
+///     compressed format if `compressed_result` is True.
 ///     """
 STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
-    mp_obj_t self, mp_obj_t secret_key, mp_obj_t public_key) {
-  const secp256k1_context *ctx = mod_trezorcrypto_get_secp256k1_context(self);
+    size_t n_args, const mp_obj_t *args) {
+  const secp256k1_context *ctx =
+      mod_trezorcrypto_get_secp256k1_context(args[0]);
   mp_buffer_info_t sk, pk;
-  mp_get_buffer_raise(secret_key, &sk, MP_BUFFER_READ);
-  mp_get_buffer_raise(public_key, &pk, MP_BUFFER_READ);
+  mp_get_buffer_raise(args[1], &sk, MP_BUFFER_READ);
+  mp_get_buffer_raise(args[2], &pk, MP_BUFFER_READ);
+  bool compressed_result = n_args == 4 && args[3] == mp_const_true;
```

Comment:
nit: this is better for future extensions 
```suggestion
  bool compressed_result = n_args >= 4 && args[3] == mp_const_true;
```

Reply:
I have reverted the change to `multiply` - the compression can be done in Python.